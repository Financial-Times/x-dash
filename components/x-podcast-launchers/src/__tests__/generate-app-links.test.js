
import generateAppLinks from '../generate-app-links';

describe('generate-app-links', () => {
	it('should generate each app link with the urlencoded RSS url', () => {
		const rssUrl = 'acast.access/rss/ft-news/&Â£@1234';
		expect(
			generateAppLinks(rssUrl)
		).toMatchSnapshot();
	})
})
