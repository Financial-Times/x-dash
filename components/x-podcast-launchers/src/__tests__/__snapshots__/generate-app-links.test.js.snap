// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-app-links should generate each app link with the urlencoded RSS url 1`] = `
Array [
  Object {
    "name": "Apple Podcasts",
    "template": "podcast://{url}",
    "url": "podcast://http%3A%2F%2Facast.access%2Frss%2Fft-news%2F%26%C2%A3%401234",
  },
  Object {
    "name": "Overcast",
    "template": "overcast://x-callback-url/add?url={url}",
    "url": "overcast://x-callback-url/add?url=http%3A%2F%2Facast.access%2Frss%2Fft-news%2F%26%C2%A3%401234",
  },
  Object {
    "name": "Pocket Casts",
    "template": "pktc://subscribe/{url}",
    "url": "pktc://subscribe/http%3A%2F%2Facast.access%2Frss%2Fft-news%2F%26%C2%A3%401234",
  },
  Object {
    "name": "Podcast Addict",
    "template": "podcastaddict://{url}",
    "url": "podcastaddict://http%3A%2F%2Facast.access%2Frss%2Fft-news%2F%26%C2%A3%401234",
  },
  Object {
    "name": "Acast",
    "template": "acast://subscribe/{url}",
    "url": "acast://subscribe/http%3A%2F%2Facast.access%2Frss%2Fft-news%2F%26%C2%A3%401234",
  },
]
`;
